web/src/lib/components/shared-components/combobox.svelte

```diff
@@ -116,8 +147,66 @@
     searchQuery = '';
     onSelect(selectedOption);
   };
+
+  const calculatePosition = (boundary: DOMRect | undefined) => {
+    dropdownDirection = getComboboxDirection(boundary);
+
+    if (!boundary) {
+      return undefined;
+    }
+
+    const vv = window.visualViewport;
+    const left = boundary.left + (vv?.offsetLeft || 0);
+    const offsetTop = vv?.offsetTop || 0;
+
+    if (dropdownDirection === 'top') {
+      return {
+        bottom: `${window.innerHeight - boundary.top - offsetTop}px`,
+        left: `${left}px`,
+        width: `${boundary.width}px`,
+        maxHeight: maxHeight(boundary.top - dropdownOffset),
+      };
+    }
+
+    const viewportHeight = vv?.height || 0;
+    const availableHeight = viewportHeight - boundary.bottom;
+    return {
+      top: `${boundary.bottom + offsetTop}px`,
+      left: `${left}px`,
+      width: `${boundary.width}px`,
+      maxHeight: maxHeight(availableHeight - dropdownOffset),
+    };
+  };
+
+  const maxHeight = (size: number) => {
+    return `min(${size}px,18rem)`;
+  };
+
+  const onPositionChange = () => {
+    if (!isOpen) {
+      return;
+    }
+    bounds = getInputPosition();
+  };
+
+  const getComboboxDirection = (boundary: DOMRect | undefined): 'bottom' | 'top' => {
+    if (!boundary) {
+      return 'bottom';
+    }
+
+    const viewportHeight = window.visualViewport?.height || 0;
+    const heightBelow = viewportHeight - boundary.bottom;
+    const heightAbove = boundary.top;
+
+    const isViewportScaled = viewportHeight && Math.floor(viewportHeight) !== Math.floor(window.innerHeight);
+
+    return heightBelow <= 225 && heightAbove > heightBelow && !isViewportScaled ? 'top' : 'bottom';
```

Comment:
Where does this 225 come from?

Reply:
I pulled this number into a variable in the code to better explain what it does.

This is the hardcoded breakpoint to decide when the dropdown should be displayed above the input box. If the available height for the dropdown is less than 225px and there's more space above than below, the dropdown will be displayed above.

225 felt to me like the right balance between not switching to the top too early, and still being able to see enough of the options when the dropdown is positioned below for it to still be useful.

Reply:
https://github.com/user-attachments/assets/01325906-1dbe-4c7b-930c-64a1addb8ffa

Here's a quick demo showing what that breakpoint looks like. 225 is the moment when the dropdown switches from below to above. 

Note that the same breakpoint applies to short dropdown menus that aren't scrolling at all - it doesn't depend on the height of the dropdown, but rather the height available.

Reply:
Awesome, thanks for clarifying it here and in the code! :))